### DISCUSSION TITLE
Issues with sending data from rp2 pico to pc via uart

### DISCUSSION THREAD

**Discussion by MariaA2005** [micropython/micropython#10517] [Using MicroPython]
I am trying to send real-time data from the rp2 pico to the pc, via uart. My code, on the rp (using Thonny), is as follows: 
```
import os
import utime
from machine import ADC

temp_sensor = ADC(4) # Default connection of temperature sensor


def temperature():
    # get raw sensor data 
    raw_sensor_data = temp_sensor.read_u16()
    
    # convert raw value to equivalent voltage
    sensor_voltage = (raw_sensor_data / 65535)*3.3
    
    # convert voltage to temperature (celcius)
    temperature = 27 - (sensor_voltage - 0.706)/0.001721
    
    return temperature
    

#print setup information :
print("OS Name : ",os.uname())

uart = machine.UART(0, baudrate = 9600)
print("UART Info : ", uart)
utime.sleep(3)

while True:
    temp = int(temperature())
    print(str(temp))
    uart.write(str(temp))
    
    
    utime.sleep(1)
```
Which gets the temperature values from the raspberry sensor and then supposedly sends them via UART. In the pc (VS Code), I have the following code :
```
  import serial
   import time
   ser = serial.Serial('COM10',9600)
  
   time.sleep(1)


while True:
    # read two bytes of data
    data = (ser.read(2))
    # convert bytestring  to unicode transformation format -8 bit
    temperature = str(data, 'UTF-8')
    print("Pico's Core Temperature : " + temperature + " Degree Celcius")
```
However, it brings up an error which says that the port is busy. Help/alternative answers?

**Comment by cve2022** [comment]
Hello, Maria.

You cannot do what you want with Thonny or any regular terminal emulator such as putty because they lock the serial port while they are running.

In your case, you may have Thonny save your Micropython script to RP2 as, say, `test_adc.py`, and create a main.py on RP2 that only `import` the script:

```
# main.py
import test_adc
```

Then quit Thonny, reset your pico, and try running the script on the VS Code side.

Short term Thonny is OK for some software development, but you should take a look at [rshell ](https://pypi.org/project/rshell/) or [mpbridge ](https://pypi.org/project/mpbridge/). There is also [mpremote](https://pypi.org/project/mpremote/), but the learning curve is a little steeper than rshell's  (I did not try mpbridge extensively yet)
